### PR TITLE
MEF: fix deleted agents

### PR DESCRIPTION
**Pull request by rerowep**
* Deleted enteties could be accessed by MEF.

Co-Authored-by: Peter Weber <peter.weber@rero.ch>

### CODE REVIEW

**Review by PascalRepond — APPROVED** [review]

File: rero_mef/api.py [hunk 1]
```diff
@@ -99,8 +99,7 @@ def create(cls, data, id_=None, delete_pid=False, dbcommit=False,
         """Create a new agent record."""
         assert cls.minter
         if '$schema' not in data:
-            type = cls.provider.pid_type
-            data = add_schema(data, type)
+            data = add_schema(data, cls.provider.pid_type)
```
Comment (jma) [inline rero_mef/api.py:102]: Should be in the commit message.

File: rero_mef/query.py [hunk 2]
```diff
@@ -60,8 +60,11 @@ def _default_parser(qstr=None):
     urlkwargs.add('q', query_string)
 
     # include deleted
-    deleted = request.args.get('deleted')
-    if not deleted:
-        search = search.filter('bool', must_not=[Q('exists', field='deleted')])
+    with_deleted = request.args.get('with_deleted')
+    if not with_deleted:
+        search = search.filter('bool', must_not=[
+            Q('exists', field='deleted'),  # no deleted MEF's
```
Comment (jma) [inline rero_mef/query.py:66]: Is it specific to MEF or for all authorities?

File: rero_mef/tasks.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def create_or_update(index, record, entity, dbcommit=True, reindex=True,
                 reindex=reindex,
                 online=online
             )
-    rec_id = returned_record.get('pid')
+    id = returned_record.get('pid')
```
Comment (jma) [inline rero_mef/tasks.py:104]: pid_value?

File: tests/unit/test_api.py [hunk 4]
```diff
@@ -62,3 +60,21 @@ def test_reromefrecord_api(app, agent_idref_record):
     process_bulk_queue(stats_only=True)
     # TODO: Find out how to test bulk indexing.
     # assert process_bulk_queue(stats_only=True) == (1, 0)
+
+
+# def test_create_deleted(agent_idref_record):
```
Comment (jma) [inline tests/unit/test_api.py:65]: To be removed?

**Review by jma — APPROVED** [review]
Please spell check your commit message.

File: rero_mef/query.py (hunk 2 above)
Comment (rerowep) [reply rero_mef/query.py:66]: For all without VIAF.

**Pull request merged**